File: articles/character-strengths-virtues-positive-psychology/cpp/character_indices.cpp
#include <iostream>

struct CharacterIndicators {
    double wisdom;
    double courage;
    double humanity;
    double justice;
    double temperance;
    double transcendence;
    double signature_strength_use;
    double authenticity_score;
    double contextual_support;
    double institutional_suppression;
    double strength_overuse_risk;
    double flourishing_score;
};

double virtue_profile_mean(const CharacterIndicators& x) {
    return (x.wisdom + x.courage + x.humanity + x.justice + x.temperance + x.transcendence) / 6.0;
}

double strength_expression_index(const CharacterIndicators& x) {
    return x.signature_strength_use + x.authenticity_score + x.contextual_support -
           x.institutional_suppression - x.strength_overuse_risk;
}

double context_adjusted_flourishing(const CharacterIndicators& x) {
    return x.flourishing_score + strength_expression_index(x) + x.contextual_support -
           x.institutional_suppression - x.strength_overuse_risk;
}

int main() {
    CharacterIndicators example{7.2, 7.0, 7.5, 7.2, 7.0, 7.4, 7.3, 7.4, 7.2, 2.9, 2.6, 7.3};

    std::cout << "Virtue profile mean: " << virtue_profile_mean(example) << "\n";
    std::cout << "Strength-expression index: " << strength_expression_index(example) << "\n";
    std::cout << "Context-adjusted flourishing: " << context_adjusted_flourishing(example) << "\n";

    return 0;
}

File: articles/character-strengths-virtues-positive-psychology/go/character_indices.go
package main

import "fmt"

type CharacterIndicators struct {
	Wisdom                   float64
	Courage                  float64
	Humanity                 float64
	Justice                  float64
	Temperance               float64
	Transcendence            float64
	SignatureStrengthUse     float64
	AuthenticityScore        float64
	ContextualSupport        float64
	InstitutionalSuppression float64
	StrengthOveruseRisk      float64
	FlourishingScore         float64
}

func VirtueProfileMean(x CharacterIndicators) float64 {
	return (x.Wisdom + x.Courage + x.Humanity + x.Justice + x.Temperance + x.Transcendence) / 6.0
}

func StrengthExpressionIndex(x CharacterIndicators) float64 {
	return x.SignatureStrengthUse + x.AuthenticityScore + x.ContextualSupport -
		x.InstitutionalSuppression - x.StrengthOveruseRisk
}

func ContextAdjustedFlourishing(x CharacterIndicators) float64 {
	return x.FlourishingScore + StrengthExpressionIndex(x) + x.ContextualSupport -
		x.InstitutionalSuppression - x.StrengthOveruseRisk
}

func main() {
	example := CharacterIndicators{7.2, 7.0, 7.5, 7.2, 7.0, 7.4, 7.3, 7.4, 7.2, 2.9, 2.6, 7.3}

	fmt.Printf("Virtue profile mean: %.3f\n", VirtueProfileMean(example))
	fmt.Printf("Strength-expression index: %.3f\n", StrengthExpressionIndex(example))
	fmt.Printf("Context-adjusted flourishing: %.3f\n", ContextAdjustedFlourishing(example))
}
